File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        dict_count = {}
        majority_count = len(nums)/2
        print(majority_count)
        for i in nums:
            current_count = dict_count.get(i,0)
            new_count = current_count+1
            if new_count > majority_count:
                return i
            dict_count[i] = current_count+1

        print(dict_count)

File: 0169-majority-element/0169-majority-element.scala
import scala.collection.mutable

object Solution {
    def majorityElement(nums: Array[Int]): Int = {
    var candidate = -1
    var counter = 0

    for (i<-nums){
    if (counter==0){
      candidate=i
    }

    if(i==candidate){
        counter+=1
    }
    else{
        counter-=1
    }

    }

    counter=0
    for (i<-nums){
      if (i==candidate){
        counter+=1
      }
    }

    if(counter>nums.length/2){
      candidate
    }else{
      -1
    

  }
    }
}